pandas/io/parquet.py

```diff
@@ -170,7 +170,7 @@ def write(
             # And pass the opened s3file to the fastparquet internal impl.
             kwargs["open_with"] = lambda path, _: path
         else:
-            path, _, _, _ = get_filepath_or_buffer(path)
+            path, _, _, _ = get_filepath_or_buffer(path, mode="wb")
```

Comment:
Can you change the `is_s3_url` on L165 to `is_s3_url(path) or is_gcs_url(path)`?